### ISSUE TITLE
Document pre-commit in CONTRIBUTING.md

### ISSUE THREAD

**Issue by mfisher87** [nsidc/earthaccess#449]

**Comment by Sherwin-14** [comment]
@mfisher87 Is this relevant? Do you have any specific opinions about this?

**Comment by mfisher87** [comment]
Yes, I think most of this is still relevant. I think this can all go in the contributing index doc? Not sure; I don't want that doc to get too big again, so open to suggestions :) Updated the original post! Let me know if there are still unanswered questions!

**Comment by Sherwin-14** [comment]
I think we should include this in contributing index doc since the topics discussed here would be in contributor interest :)

**Comment by Sherwin-14** [comment]
Hey should we assign an entirely different section addressing these points? I wonder which is the section where all of these fit together

**Comment by mfisher87** [comment]
Do you mean a section within the [contributing index doc](https://github.com/nsidc/earthaccess/blob/main/docs/contributing/index.md)? We could update the "format and lint" section to use pre-commit instead of Make.

Perhaps for the docs stuff, we add a subsection "... to contribute to docs"?

Perhaps for conda v poetry, we could add a section for "development environment management". Where do we tell contributors how to use poetry to install a development environment? I don't know how personally (and I think we should switch away from poetry ;) ). I don't know that we actually document that currently, so I think we might need new content.

**Comment by Sherwin-14** [comment]
Yeah I mean sections under contributing index doc. I guess it's best to address each of the points above by creating their own respective sections, but then this might make that doc big so would that be permissible? Maybe we can put the poetry issue on hold for now and once things are sorted then we can add the add it later.

**Comment by mfisher87** [comment]
I think that sounds like a great path forward, thank you @Sherwin-14 :) Would you like to be assigned to this issue?

**Comment by Sherwin-14** [comment]
Yes I would be glad :)

**Comment by saberbrasher** [comment]
Hi, folks! there have been a lot of changes to our contributing docs since this issue was last touched. Are there points here that we still need to address?

**Comment by mfisher87** [comment]
pre-commit: https://earthaccess.readthedocs.io/en/stable/contributor/development/#usage-of-pre-commit

google-style docstring: https://earthaccess.readthedocs.io/en/stable/contributor/development/#documentation-style

How to work on docs locally: Same as above

Hot reloader limitations: Same as above!

Thanks for pointing this out! Closing.
